shared/health/src/main/java/io/pravega/shared/health/Details.java

```diff
@@ -0,0 +1,16 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.health;
+
+import java.util.HashMap;
+
+
+public class Details extends HashMap<String, String> {
```

Comment:
Why do you need a class that just extends from HashMap? Can't you just return `Map<String, String>` which enables the supplier to provide their own implementation of `Map`? (Such as immutable map?)